airflow/providers/opensearch/hooks/opensearch.py

```diff
@@ -0,0 +1,122 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import json
+from functools import cached_property
+from typing import Any
+
+from opensearchpy import OpenSearch, RequestsHttpConnection
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+
+
+class OpenSearchHook(BaseHook):
+    """
+    Provide a thin wrapper around the OpenSearch client.
+
+    :param: open_search_conn_id: Connection to use with Open Search
+    :param: log_query: Whether to log the query used for Open Search
+    """
+
+    conn_name_attr = "opensearch_conn_id"
+    default_conn_name = "opensearch_default"
+    conn_type = "opensearch"
+    hook_name = "OpenSearch Hook"
+
+    def __init__(self, open_search_conn_id: str, log_query: bool, **kwargs: Any):
+        super().__init__(**kwargs)
+        self.conn_id = open_search_conn_id
+        self.log_query = log_query
+
+        self.use_ssl = self.conn.extra_dejson.get("use_ssl", False)
+        self.verify_certs = self.conn.extra_dejson.get("verify_certs", False)
+        self.__SERVICE = "es"
+
+    @cached_property
+    def conn(self):
+        return self.get_connection(self.conn_id)
+
+    @cached_property
+    def get_client(self) -> OpenSearch:
```

Comment:
nit: since it is a property, I would just call it `client`
```suggestion
    def client(self) -> OpenSearch:
```

Reply:
There was some miscommunication via the thread on the branches showing up from creating this branch not from mainline. So I tried to be proactive and drop those commits from this branch which ideally I don’t want to push that change unless it was absolutely necessary. I can clean this up if it is blocking, the suggestion you have can’t be applied given the references to this method in other places in the code.

Reply:
I had to push changes for breeze commands so I am pushing this change now.